### PR TITLE
feat: shared types

### PR DESCRIPTION
**Pull request by denkristoffer**
# Purpose of PR

Adds common types to `@contentful/f36-core`. This could potentially be expanded to organization wide types at a later point.

### CODE REVIEW

File: packages/forma-36-react-components/src/components/EntityList/EntityListItem/EntityListItem.tsx [hunk 1]
```diff
@@ -13,11 +14,12 @@ import {
...
 import styles from './EntityListItem.css';
 
-export type EntityListItemStatus =
-  | 'archived'
-  | 'changed'
-  | 'draft'
-  | 'published';
+type EntityListItemStatusKeys = 'archived' | 'changed' | 'draft' | 'published';
+
+type EntityListItemStatus = Exclude<
```
[the remark this answers is not in the document]
Comment (denkristoffer) [reply packages/forma-36-react-components/src/components/EntityList/EntityListItem/EntityListItem.tsx:19]: `Omit` only works with type objects, not with enums, same with `Pick`. I thought about wrapping the double `Exclude` in a utility type, would that work for you?

**Review by gui-santos — COMMENTED** [review]
Looks cool to me!

I have only one question:

Since we are moving components to their own packages, is this change going to affect how people will use types of some components?

For example:

if someone is doing this
```tsx
import { Button, ButtonVariants } from '@contentful/f36-button'
```
will they need to do this instead?
```tsx
import { Variants } from '@contentful/f36-types'
import { Button } from '@contentful/f36-button'
```

**Comment by denkristoffer** [comment]
@gui-santos That's a good point, I think we should re-export from the component packages. My intention was, for now, for this to be an internal package.

**Comment by denkristoffer** [comment]
I'm learning about a lot of TypeScript compilation quirks with this PR! 🤯

**Review by suevalov — COMMENTED** [review]
After thinking of it one more time, I suggest keeping all shared types in `f36-core` package. There won't be many of them and it makes sense to collocate all core things in one package.

**Comment by denkristoffer** [comment]
@suevalov I agree, we can land this after the core package.

**Comment by denkristoffer** [comment]
Working on simplyfing this to just strings. Even though I like enums there are also a number of compilation issues with them that make it too bothersome to use them.

**Review by gui-santos — APPROVED** [review]
great stuff!

**Review by mshaaban0 — APPROVED** [review]
🔥🖤

**Review by suevalov — APPROVED** [review]

**Pull request merged**